offload/plugins-nextgen/common/CMakeLists.txt

```diff
@@ -1,13 +1,16 @@
 # NOTE: Don't try to build `PluginInterface` using `add_llvm_library` because we
 # don't want to export `PluginInterface` while `add_llvm_library` requires that.
-add_library(PluginCommon OBJECT
+add_library(PluginCommon STATIC
```

Comment:
Why is this static now?

Reply:
As I've noted in the commit message, `llvm_config()` doesn't support object libraries.

Reply:
I'm pretty sure that making this static will break something, but I can't remember what it is exactly. The LLVM configs here should probably be done when we make the plugin static library, `add_target_library`.

Reply:
Well, all tests passed for me with this change — but I can alternatively change `llvm_config` to use `PRIVATE` linking for object libraries, but I don't know if that isn't going to break something else.

Reply:
I don't think the LLVM profile data thing should go here at least. Maybe @EthanLuisMcDonough knows if it was necessary to put it there. All of the LLVM libs should be handled in `add_target_library`, however we do need the dependency on intrinsics because we depend on it.

This is intentionally supposed to create a single object library that's then used to create the static libraries, we could maybe do it with static but I don't see the point when this isn't where any LLVM logic should go anyway.